File: app/src/main/java/br/com/disapps/homepet/ui/hotel/HotelActivity.java
package br.com.disapps.homepet.ui.hotel;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.constraint.ConstraintLayout;
import android.support.design.widget.AppBarLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.FrameLayout;

import com.rd.PageIndicatorView;

import java.util.ArrayList;
import java.util.List;

import br.com.disapps.homepet.R;
import br.com.disapps.homepet.app.HomePet;
import br.com.disapps.homepet.data.model.Hotel;
import br.com.disapps.homepet.ui.common.AppActivity;
import br.com.disapps.homepet.ui.custom.LoadingView;
import br.com.disapps.homepet.ui.hotel.hotelFragment.HotelFragment;
import br.com.disapps.homepet.ui.hotels.adapter.ImageViewPagerAdapter;
import butterknife.BindView;

/**
 * Created by diefferson.santos on 31/08/17.
 */

public class HotelActivity  extends AppActivity
        implements AppBarLayout.OnOffsetChangedListener, IHotelView {

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @BindView(R.id.image_slide)
    ViewPager bannerSlider;

    @BindView(R.id.container)
    FrameLayout container;

    @BindView(R.id.pageIndicatorView)
    PageIndicatorView pageIndicatorView;

    @BindView(R.id.loading_view)
    ConstraintLayout loadingView;

    private HotelPresenter hotelPresenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_hotel);
        bindView();

        setToolbar(toolbar);
        setContainer(container);

        int hotelCode = getIntent().getExtras().getInt("hotelCode");

        getPresenter().loadHotel(hotelCode);

        replaceFragment(HotelFragment.newInstance(hotelCode));

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        int id = item.getItemId();

        switch (id){
            case android.R.id.home:
                onBackPressed();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void showLoading(boolean cancelable) {
        loadingView.setVisibility(View.VISIBLE);
    }

    @Override
    public void dismissLoading() {
        loadingView.setVisibility(View.GONE);
    }

    public void fillHeaderHotel(Hotel hotel) {

        setTitle(hotel.getName());

        bannerSlider.setAdapter(new ImageViewPagerAdapter(this,hotel.getImages() ));
        pageIndicatorView.setViewPager(bannerSlider);
        pageIndicatorView.setVisibility(View.VISIBLE);

    }

    private HotelPresenter getPresenter(){

        if(hotelPresenter == null){
            createPresenter();
        }

        return hotelPresenter;
    }

    private void createPresenter() {
        hotelPresenter =  new HotelPresenter(HomePet.Companion.getInstance().getHotelRepository());
        hotelPresenter.attachView(this);
    }

    @Override
    public void onOffsetChanged(AppBarLayout appBarLayout, int verticalOffset) {

    }

}

File: app/src/main/java/br/com/disapps/homepet/ui/hotel/HotelActivity.kt
package br.com.disapps.homepet.ui.hotel

import android.content.Intent
import android.opengl.Visibility
import android.os.Bundle
import android.support.design.widget.AppBarLayout
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import br.com.disapps.homepet.R
import br.com.disapps.homepet.app.HomePet
import br.com.disapps.homepet.data.model.Hotel
import br.com.disapps.homepet.ui.common.AppActivity
import br.com.disapps.homepet.ui.hotel.hotelFragment.HotelFragment
import br.com.disapps.homepet.ui.hotels.adapter.ImageViewPagerAdapter
import br.com.disapps.homepet.ui.imageViewer.ImageViewerActivity
import br.com.disapps.homepet.ui.profile.ProfileFragment
import kotlinx.android.synthetic.main.activity_hotel.*
import org.jetbrains.anko.startActivity

/**
 * Created by diefferson.santos on 31/08/17.
 */

class HotelActivity : AppActivity(), AppBarLayout.OnOffsetChangedListener, IHotelView {

    private var hotelPresenter: HotelPresenter? = null
    private var hotel: Hotel?  = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_hotel)

        setToolbar(toolbar)
        setContainer(container!!)

        val hotelCode = intent.extras!!.getInt("hotelCode")

        presenter.loadHotel(hotelCode)

        replaceFragment(HotelFragment.newInstance(hotelCode))

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        val id = item.itemId

        when (id) {
            android.R.id.home -> {
                onBackPressed()
                return true
            }
            R.id.full_screem->{startActivity<ImageViewerActivity>("hotel" to hotel!!) }
        }

        return super.onOptionsItemSelected(item)
    }

    override fun fillHeaderHotel(h: Hotel) {

        hotel = h

        setTitle(hotel!!.name!!)

        if(hotel!!.images != null && !hotel!!.images!!.isEmpty()){
            image_slide!!.adapter = ImageViewPagerAdapter(this, hotel!!.images!!)
            pageIndicatorView!!.setViewPager(image_slide)
            pageIndicatorView!!.visibility = View.VISIBLE
            image_slide.visibility = View.VISIBLE

            collapsing.setOnClickListener {  startActivity<ImageViewerActivity>("hotel" to hotel!!) }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.hotel_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    private val presenter: HotelPresenter
        get() {

            if (hotelPresenter == null) {
                createPresenter()
            }

            return hotelPresenter!!
        }

    private fun createPresenter() {
        hotelPresenter = HotelPresenter(HomePet.instance!!.hotelRepository!!)
        hotelPresenter!!.attachView(this)
    }

    override fun onOffsetChanged(appBarLayout: AppBarLayout, verticalOffset: Int) {

    }

}